Library/Homebrew/diagnostic.rb

```diff
@@ -535,6 +535,24 @@ def check_user_path_3
         EOS
       end
 
+      def check_xdg_data_dirs
+        share = "#{HOMEBREW_PREFIX}/share"
+        homebrew_in_xdg_data_dirs = true
+        if ENV.key?("XDG_DATA_DIRS")
+          xdg_data_dirs = ENV["XDG_DATA_DIRS"].split(File::PATH_SEPARATOR)
+          xdg_data_dirs.length > 0 && homebrew_in_xdg_data_dirs = xdg_data_dirs.include?(share)
```

Comment:
One of the tests tests (sic!) this case. When the variable is set but it is empty we get the warning which should not be there as `vapigen` will work with it.

Your approach is more concise, then. It does no harm, it simply is more strict than it needs to be.